extension/src/time_vector.rs

```diff
@@ -424,6 +477,99 @@ mod tests {
         })
     }
 
+    #[pg_test]
+    pub fn test_format_timevector() {
+        Spi::execute(|client| {
+            client.select("SET timezone TO 'UTC'", None, None);
+            client.select(
+                "CREATE TABLE data(time TIMESTAMPTZ, value DOUBLE PRECISION)",
+                None,
+                None,
+            );
+            client.select(
+                r#"INSERT INTO data VALUES
+                    ('2020-1-1', 30.0),
+                    ('2020-1-2', 45.0),
+                    ('2020-1-3', NULL),
+                    ('2020-1-4', 55.5),
+                    ('2020-1-5', 10.0)"#,
+                None,
+                None,
+            );
+
+            let test_plotly_template = client.select(
+                "SELECT toolkit_experimental.format_timevector(timevector(time, value),'plotly') FROM data",
+                None,
+                None,
+            ).first()
+                .get_one::<String>()
+                .unwrap();
+
+            assert_eq!(test_plotly_template,"{\"times\": [\n  \"2020-01-01 00:00:00+00\",\n  \"2020-01-02 00:00:00+00\",\n  \"2020-01-03 00:00:00+00\",\n  \"2020-01-04 00:00:00+00\",\n  \"2020-01-05 00:00:00+00\"\n], \"vals\": [\n  \"30\",\n  \"45\",\n  \"null\",\n  \"55.5\",\n  \"10\"\n]}"
+		     );
+
+            let test_paired_template = client.select(
+                "SELECT toolkit_experimental.format_timevector(timevector(time, value),'paired') FROM data",
+                None,
+                None,
+            ).first()
+                .get_one::<String>()
+                .unwrap();
+
+            assert_eq!(
+                test_paired_template,
+        	"[{\"time\": \"2020-01-01 00:00:00+00\", \"val\": 30}, {\"time\": \"2020-01-02 00:00:00+00\", \"val\": 45}, {\"time\": \"2020-01-03 00:00:00+00\", \"val\": null}, {\"time\": \"2020-01-04 00:00:00+00\", \"val\": 55.5}, {\"time\": \"2020-01-05 00:00:00+00\", \"val\": 10} ]"
+            );
+
+            let test_user_supplied_template = client.select(
+                "SELECT toolkit_experimental.format_timevector(timevector(time, value),'{\"times\": {{ TIMES | json_encode(pretty=true) | safe  }}, \"vals\": {{ VALUES | json_encode(pretty=true) | safe }}}') FROM data",
```

Comment:
Is the `json_encode(pretty=true) | safe` required to make this look like valid JSON, can we do that by default (probably not `pretty=true` by default though)?

Reply:
json_encode() is required but `pretty=true` isn't so I can turn that off if we want to.
If you were to write this instead of using json_encode:
```{\"times\": {{ TIMES }}, \"vals\": {{ VALUES }}}``` it would render as:
```
`"{\"times\": [2020-01-01 00:00:00+00, 2020-01-02 00:00:00+00, 2020-01-03 00:00:00+00, 2020-01-04 00:00:00+00, 2020-01-05 00:00:00+00], \"vals\": [30, 45, null, 55.5, 10]}"
```
json_encode without pretty printing:
```
{\"times\": {{ TIMES | json_encode() | safe  }}, \"vals\": {{ VALUES | json_encode() | safe }}}'
``` looks like this:
```
"{\"times\": [\"2020-01-01 00:00:00+00\",\"2020-01-02 00:00:00+00\",\"2020-01-03 00:00:00+00\",\"2020-01-04 00:00:00+00\",\"2020-01-05 00:00:00+00\"], \"vals\": [\"30\",\"45\",\"null\",\"55.5\",\"10\"]}"`
```
There doesn't seem to be a good builtin way to run json_encode() on all user template code by default though. This might be something we can investigate in a future improvement?